### PR TITLE
Add home volume mount for che-theia editor

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add home folder volume mount for che-theia editor. The GitHub authentication session is saved in this directory, so the GitHub authentication will be restored after workspace restart.

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/18782

### How to test this PR?

Open the che-theia's editor `meta.yaml` (< plugin registry url >/v3/plugins/eclipse/che-theia/next/) and see the workspace storage volume mount:
```
      volumes:
        - name: plugins
          mountPath: /plugins
        - name: home
          mountPath: /home/theia/.theia
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by benoitf** [comment]
Hello, looks like happy path is not started
Pattern used looks incorrect there
https://github.com/eclipse/che-plugin-registry/blob/master/.github/workflows/happy-path.yaml#L17-L22

Could you try to modify the pattern to be `**/*` instead of `.github/workflows/happy-path.yaml`

File: .github/workflows/happy-path.yaml [hunk 1]
```diff
@@ -14,12 +14,10 @@ name: Happy Path
 on:
   pull_request:
     paths:
-      - '.github/workflows/happy-path.yaml'
-      - '!sidecars/**'
+      - '**/*'
```
Comment (benoitf) [inline .github/workflows/happy-path.yaml:17]: can you keep - '!sidecars/**' ?

File: che-editors.yaml [hunk 2]
```diff
@@ -77,6 +77,8 @@ editors:
         volumeMounts:
           - name: plugins
             path: "/plugins"
+          - name: home
```
Comment (sunix) [inline che-editors.yaml:80]: I am not sure `home` is the right name to give. Users may use that name for a volume for the user home (`/home/theia` or `/home/user`). I would name it `theia-local` or something like that.

**Review by sunix — CHANGES_REQUESTED** [review]

File: .github/workflows/happy-path.yaml (hunk 1 above)
Comment (vinokurig) [reply .github/workflows/happy-path.yaml:17]: done

File: che-editors.yaml (hunk 2 above)
Comment (sunix) [reply che-editors.yaml:80]: `theia-local-config`

Comment (vinokurig) [reply che-editors.yaml:80]: fixed

Comment (sunix) [reply che-editors.yaml:80]: or `theia-local-storage`

Comment (vinokurig) [reply che-editors.yaml:80]: I would keep `theia-local` if there is no objections?

Comment (sunix) [reply che-editors.yaml:80]: @vinokurig I can see only one of the 2 files updated

Comment (vinokurig) [reply che-editors.yaml]: @sunix updated the secnd one, thanks.

**Review by sunix — DISMISSED** [review]
code looks good to me

**Comment by sunix** [comment]
@vinokurig I would like to check if `/home/theia/.theia/settings.json` is affected

**Comment by ericwill** [comment]
Code looks good to me, but Happy Path from PR is failing

**Comment by benoitf** [comment]
@ericwill happy path test is not so stable so if there is at least one Happy Path that is passing, that's perfect :-)

**Review by sunix — APPROVED** [review]
OK i tried by just setting the volume from the devfile and it is all fine !

**Pull request merged**